app/src/main/java/nextstep/shoppingcart/ui/view/product/list/ProductListScreen.kt

```diff
@@ -49,21 +53,33 @@ fun ProductListScreen() {
             )
         },
     ) { paddingValues ->
+        var cartItems by remember {
+            mutableStateOf(Cart.items)
+        }
         LazyVerticalGrid(
             modifier = Modifier.padding(paddingValues = paddingValues),
             columns = GridCells.Fixed(2),
             contentPadding = PaddingValues(18.dp),
             verticalArrangement = Arrangement.spacedBy(20.dp),
             horizontalArrangement = Arrangement.spacedBy(12.dp)
         ) {
+
             items(
                 items = dummyProducts,
                 key = { product -> product.name }
             ) {
                 ProductListItem(
                     product = it,
+                    countInCart = cartItems.find { cartItem -> cartItem.product == it }?.count ?: 0,
                     onClick = { product ->
                         context.startActivity(ProductDetailActivity.newIntent(context, product))
+                    },
+                    onQuantityChange = { cartItem, newQuantity ->
+                        cartItems = if (cartItem.count > newQuantity) {
+                            Cart.removeOne(cartItem.product)
+                        } else {
+                            Cart.addOne(cartItem.product)
+                        }
```

Comment:
가끔은 이렇게 when절로 쓰면 기분이가 좋더라구요.. 라인 수는 큰 차이가 없는데 말이죠 ㅋㅋ
```suggestion
                        cartItems = when (cartItem.count > newQuantity) {
                            true -> Cart.removeOne(cartItem.product)
                            false -> Cart.addOne(cartItem.product)
                        }
```

Reply:
저도 저렇게 쓰던 시절이 있었는데 Boolean은 when보다 if로 쓰는게 손이 더 가요 :)